### DISCUSSION TITLE
Not able to serve bentoml API on pre-trained .h5 file

### DISCUSSION THREAD

**Discussion by arpitkjain10** [bentoml/BentoML#1169] [General]
Hello!
Using bentoml to serve my tensorflow model. added my pre-trained sequential model to artifacts using beentoml decorator.
When triggering the API created from swagger, getting invalid model error.
Error: "Exception happened in API function: '_UserObject' object has no attribute 'predict'"
Code: prediction = self.artifacts.model.predict(testing_padded_seq)
I checked and got to know that the above line is laoding a UserObject.
<tensorflow.python.saved_model.load.Loader._recreate_base_user_object.<locals>._UserObject object at 0x7fb82c65af90>
Completely stuck. Please let me know if anyone know the solution here.

**Comment by yubozhao** [comment]
hi @arpitkjain10 

For BentoML, it packages the model and all of the dependencies.  It doesn't add additional methods on to model itself.

You will need to check your model framework, what method is available for the model to call.

For tensorflow, I don't think `predict` method is available for the model.  In the [BentoML gallery repo](https://github.com/bentoml/gallery), you can find the tensorflow example notebooks that is calling model itself
```python
 @bentoml.api(input=TfTensorInput(), batch=True)
    def predict(self, tensor):
        outputs = self.artifacts.model(tensor)
        return outputs
```
In notebook: https://github.com/bentoml/gallery/blob/master/tensorflow/echo/tensorflow-echo.ipynb

Hope this helps, and let me know how it goes, I am happy to help

**Reply by arpitkjain10** [reply]
Hi @yubozhao 

The suggestion you made worked for me. I converted the array into tensor and passed as input to self.artifacts.model.
Thanks for the help. 👍
